tools/extra/ras/main.c

```diff
@@ -896,27 +896,63 @@ fpga_result inject_ras_errors(fpga_handle fme_handle,
 	result = fpgaObjectWrite64(inj_error_object, inj_error.csr, 0);
 	if (result != FPGA_OK) {
 		OPAE_ERR("Failed to Read Object ");
-		return result;
+		goto out;
 	}
-
+out:
 	result = fpgaDestroyObject(&inj_error_object);
 	if (result != FPGA_OK) {
 		OPAE_ERR("Failed to Destroy Object");
-		return result;
 	}
 
 	printf("----------- INJECT ERROR  END-------- \n \n");
 	return result;
 }
 
+static fpga_result clear_fpga_errors(fpga_handle handle, char *error )
+{
+	fpga_result result = FPGA_OK;
+	fpga_object error_object;
+	uint64_t value; 
+
+	if (!error)
+		return FPGA_INVALID_PARAM;
+
+	// Clear FME error
+	result = fpgaHandleGetObject(handle, error, &error_object, 0);
+	if (result != FPGA_OK) {
+		OPAE_ERR("Failed to get handle Object");
+		return result;
+	}
+
+	result = fpgaObjectRead64(error_object, &value, 0);
+	if (result != FPGA_OK) {
+		OPAE_ERR("Failed to Read Object ");
+		goto out;
+	}
+
+	result = fpgaObjectWrite64(error_object, value, 0);
+	if (result != FPGA_OK) {
+		OPAE_ERR("Failed to Write Object ");
+		goto out;
```

Comment:
You lose the value of `result` when you `goto out`. Also not needed here since we can just fall through.